cocotb/__init__.py

```diff
@@ -187,6 +187,11 @@ def _reopen_stream_with_buffering(stream_name):
 
 def fork(coro: Union[RunningTask, Coroutine]) -> RunningTask:
     """Schedule a coroutine to be run concurrently. See :ref:`coroutines` for details on its use."""
+    warnings.warn(
+        "fork has been deprecated in favor of cocotb.start_soon and cocotb.start",
+        DeprecationWarning,
+        stacklevel=2,
```

Comment:
Deprecation warnings, IMO, should not only tell what's wrong, but also give actionable (ideally, copy/paste) advice how to change to the new behavior.

For that, we should suggest a single preferred replacement for fork, instead of giving start_soon and start as options. https://www.fossi-foundation.org/2021/10/20/cocotb-1-6-0 says: "In most cases, you can simply replace cocotb.fork() with cocotb.start_soon().". Something like that would be in order here as well.

We could also add this blog post as link to the error message to help people decide if they need start() instead. (I wanted for a while now move this content into the documentation. I'll update the link once we do that.)

Reply:
Added a longer and more descriptive message.